CONTRIBUTING.md

```diff
@@ -21,14 +21,90 @@ Unsure where to begin contributing? You can start by looking through these `begi
+  - Clone the project locally into your system.
+```
+git clone https://github.com/your-username/bench-routes.git
+```
+  - Make sure you are in the `master` branch.
+```
+git checkout master
+```
+  - Create a new branch with a meaningful name for each feature/pull request. 
+```
+git checkout -b branch-name
+```
+  - Make all the required changes.
+
+  - Ensure that the code follows the linting standards. `eslint`, `tslint` and `prettier` are used for the TypeScript files and `golint` is used for the Golang files. Use the command `make fix` to format the code.
+  - Add the files you changed.
+```
+git add file-name
+```
+  - Write appropriate comments in the pull request, explaining the changes made. This helps maintainers, and the community understand the changes made by you and help review the PR. **Reference the issue you are fixing**.
+```
+git commit -m "message"
+```
+  - If you forgot to add some changes, you can edit your previous commit message.
+```
+git commit --amend
+```
+  - Squash multiple commits to a single commit. (example: squash last two commits done on this branch into one)
+```
+git rebase --interactive HEAD~2 
+```
+  - Push this branch to your remote repository on GitHub.
+```
+git push origin branch-name
+```
+  - If any of the squashed commits have already been pushed to your remote repository, you need to do a force push.
+```
+git push origin remote-branch-name --force
+```
+
+  - For changes to the front-end include screenshots or screencast showing the working changes. You can use browser extensions like `Screencastify` to create a screencast.
+
+  - Make sure you submit some kind of proof of your fix that makes the reviewing process easier for the maintainers.
+
+  - During review, if you are requested to make changes, rebase your branch and squash the multiple commits into one. Once you push these changes the pull request will edit automatically.
+
+## Configuring remotes
+When a repository is cloned, it has a default remote called `origin` that points to your fork on GitHub, not the original repository it was forked from. To keep track of the original repository, you should add another remote called `upstream`.
+
+1. Set the `upstream`.
+```
+git remote add upstream https://github.com/bench-routes/bench-routes.git
+```
+2. Use `git remote -v` to check the status. The output must be something like this:
+```
+  > origin    https://github.com/your-username/bench-routes.git (fetch)
+  > origin    https://github.com/your-username/bench-routes.git (push)
+  > upstream  https://github.com/bench-routes/bench-routes.git (fetch)
+  > upstream  https://github.com/bench-routes/bench-routes.git (push)
+```
+3. To update your local copy with remote changes, run the following: (This will give you an exact copy of the current remote. You should not have any local changes on your master branch, if you do, use rebase instead.)
+```
+git pull upstream master
```

Comment:
This will cause confusion. We should only say the fetch and rebasing method, as that works for all cases.

Reply:
Yes, I have mentioned that point in Line number 101, this is for updating the master branch in the forked repo.